src/fmeval/transforms/common.py

```diff
@@ -41,82 +40,55 @@ def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+
+            Note that the reason a list of output keys is used (as opposed to
+            a singular key) is so that `model_runner` can be invoked on the
+            same input multiple times.
+
+            Note that the response payload from calling `predict` will be a tuple of the
+            form (model_output, log_probability), and this transform is only concerned with
+            the model_output element.
+        :param model_runner: The ModelRunner instance whose outputs will be obtained.
         """
-        super().__init__(input_key_to_response_keys, model_runner)
+        super().__init__(input_to_output_keys, model_runner)
         self.register_input_output_keys(
-            input_keys=list(input_key_to_response_keys.keys()),
+            input_keys=list(input_to_output_keys.keys()),
             output_keys=[
-                response_key
-                for list_of_response_key_tuples in input_key_to_response_keys.values()
-                for response_key_tuple in list_of_response_key_tuples
-                for response_key in response_key_tuple
+                output_key for output_key_list in input_to_output_keys.values() for output_key in output_key_list
             ],
         )
-        self.input_key_to_response_keys = input_key_to_response_keys
+        self.input_to_output_keys = input_to_output_keys
         self.model_runner = model_runner
 
     @validate_call
     def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
-        """Augment the input record with model responses and returns said record.
+        """Augment the input record with model outputs and return said record.
 
         :param record: The input record.
-        :returns: The input record with model response data added in.
+        :returns: The input record with model output data added in.
         """
-        for input_key in self.input_keys:
-            response_key_tuples = self.input_key_to_response_keys[input_key]
-            for response_key_tuple in response_key_tuples:
-                model_output, log_prob = self.model_runner.predict(record[input_key])
-                model_response = ((model_output,) if model_output is not None else ()) + (
-                    (log_prob,) if log_prob is not None else ()
-                )
-                assert_condition(
-                    len(model_response) == len(response_key_tuple),
-                    f"The number of elements in model response {model_response} "
-                    f"does not match number of response keys in {response_key_tuple}.",
-                )
-                for model_response_key, model_response_item in zip(response_key_tuple, model_response):
-                    record[model_response_key] = model_response_item
+        for input_key, output_keys in self.input_to_output_keys.items():
+            for output_key in output_keys:
+                model_output, _ = self.model_runner.predict(record[input_key])
```

Comment:
Great! Next step is to refactor also the interface of `ModelRunner` to separate the two calls. 
(and also get a transfrom for input log prob that is used in Stereotyping)